src/index.ts

```diff
@@ -1,140 +1,42 @@
 	const logger = new Logger(botConfig);
-	let getFlashArbMessages = chains.defaults.getFlashArbMessages;
-	let getPoolStates = chains.defaults.getPoolStates;
-	let initPools = chains.defaults.initPools;
-	let startupTime = Date.now();
-	let timeIt = 0;
 
-	await import("./chains/" + botConfig.chainPrefix).then(async (chainSetups) => {
-		if (chainSetups === undefined) {
-			await logger.sendMessage("Unable to resolve specific chain imports, using defaults", LogType.Console);
-		}
-		getFlashArbMessages = chainSetups.getFlashArbMessages;
-		getPoolStates = chainSetups.getPoolStates;
-		initPools = chainSetups.initPools;
-		return;
-	});
+	// print the config
+	await logger.defaults.logConfig(botConfig);
+	//spawn chainOperator for interaction with blockchains
 	const chainOperator = await ChainOperator.connectWithSigner(botConfig);
-	let setupMessage = "---".repeat(30);
-	const allPools = await initPools(chainOperator, botConfig.poolEnvs, botConfig.mappingFactoryRouter);
-	const graph = newGraph(allPools);
-	const paths = getPaths(graph, botConfig.offerAssetInfo, botConfig.maxPathPools) ?? [];
-
-	const filteredPools = removedUnusedPools(allPools, paths);
-	setupMessage += `**\nDerived Paths for Arbitrage:
-Total Paths:** \t${paths.length}\n`;
-	for (let pathlength = 2; pathlength <= botConfig.maxPathPools; pathlength++) {
-		const nrOfPaths = paths.filter((path) => path.pools.length === pathlength).length;
-		setupMessage += `**${pathlength} HOP Paths:** \t${nrOfPaths}\n`;
-	}
-
-	setupMessage += `(Removed ${allPools.length - filteredPools.length} unused pools)\n`;
-	setupMessage += "---".repeat(30);
-
-	startupMessage += setupMessage;
-	await logger.sendMessage(startupMessage, LogType.Console);
-
+	//create the arbitrage loop based on input config
 	let loop;
-
-	if (botConfig.skipConfig) {
-		await logger.sendMessage("Initializing skip loop...", LogType.Console);
-		const [skipClient, skipSigner] = await getSkipClient(
-			botConfig.skipConfig.skipRpcUrl,
-			botConfig.mnemonic,
-			botConfig.chainPrefix,
-		);
-		loop = new SkipLoop(
-			filteredPools,
-			paths,
-			trySomeArb,
-			getPoolStates,
-			getFlashArbMessages,
-			chainOperator,
-			botConfig,
-			skipClient,
-			skipSigner,
-			logger,
-			[...paths],
-			botConfig.ignoreAddresses,
-		);
-	} else if (botConfig.useMempool === true) {
-		await logger.sendMessage("Initializing mempool loop...", LogType.Console);
-
-		loop = new MempoolLoop(
-			filteredPools,
-			paths,
-			trySomeArb,
-			getPoolStates,
-			getFlashArbMessages,
-			chainOperator,
-			botConfig,
-			logger,
-			[...paths],
-			botConfig.ignoreAddresses,
-		);
-	} else {
-		await logger.sendMessage("Initializing non-mempool loop...", LogType.Console);
-		loop = new NoMempoolLoop(
-			filteredPools,
-			paths,
-			trySomeArb,
-			getPoolStates,
-			getFlashArbMessages,
-			chainOperator,
-			botConfig,
-			logger,
-			[...paths],
-		);
+	if (botConfig.setupType === SetupType.DEX) {
```

Comment:
case usable?

`switch (botConfig.setupType) {`

		case SetupType.DEX:
			loop = await DexLoop.createLoop(chainOperator, <DexConfig>botConfig, logger);
			//print the created arbitrage loop
			await logger.defaults.logDexLoop(loop)
			break;
		case SetupType.LIQUIDATION:
			loop = await LiquidationLoop.createLoop(chainOperator, <LiquidationConfig>botConfig, logger);
			//print the created liquidation loop
			await logger.defaults.logLiqLoop(loop);
			break;
	}
`

Reply:
done, nice addition